Adding docs for pod capabilities feature flat (see serving PR #11410) (#3666)
knative/docs: docs/serving/feature-flags.md

        ...
```

## Kubernetes Security Context Capabilities

* **Type**: extension
* **ConfigMap key**: `kubernetes.podspec-addcapabilities`

> This flag controls whether users can add capabilities on the Pod's securityContext.

Reviewer: ```suggestion
This flag controls whether users can add capabilities on the `securityContext` of a pod.
```

> When set to "enabled" or "allowed" it allows [Linux capabilities](https://man7.org/linux/man-pages/man7/capabilities.7.html) to be added to the container.

Reviewer: ```suggestion
When set to `enabled` or `allowed` it allows [Linux capabilities](https://man7.org/linux/man-pages/man7/capabilities.7.html) to be added to the container.
```

Is enabled / allowed a literal setting? Where does someone switch this on/off?

Author: This is set in the [`config-features` configmap](https://github.com/psschwei/serving/blob/add-add-capability/config/core/configmaps/features.yaml#L112) in the `knative-serving` namespace. Definitions for the enabled / allowed values are provided in the [`docs/feature-flags.md`](https://github.com/knative/docs/blob/mkdocs/docs/serving/feature-flags.md#control) page.

On `enabled` vs. "enabled" -- happy to make that change. There's a number of other places in the doc where the values are in quotes, would it be better to update them all at once in a separate PR (maybe together with the yaml examples mentioned in a previous comment)?

Reviewer: > This is set in the config-features configmap in the knative-serving namespace. Definitions for the enabled / allowed values are provided in the docs/feature-flags.md page.

I'd at least call that out and link to that page maybe, as someone reading this I wouldn't know how to enable it in the configmap without the pointer.

Author: Sorry, I got the link wrong... those definitions are actually on this same page (in the first section).

```yaml
apiVersion: serving.knative.dev/v1
kind: Service
...
spec:
 template:
  spec:
   containers:
      securityContext:
        capabilities:
          add:
            - NET_BIND_SERVICE
            ...

> ```

Reviewer: Can we make this a full sample just to avoid confusion? It seems truncated?

Author: A full sample is okay with me. Though it looks like most of the other yaml examples in this doc are also snippets rather than full samples -- would it be better to make a separate PR to update all of them?
